test/MathOptInterface/MOI_callbacks.jl

```diff
@@ -251,123 +327,124 @@ end
-                        0.0
-                    ),
-                    MOI.LessThan{Float64}(1.0)
-                )
-            elseif y_val + x_val > 3 + 1e-6
-                MOI.submit(model, MOI.LazyConstraint(cb_data),
-                    MOI.ScalarAffineFunction{Float64}(
-                        MOI.ScalarAffineTerm.([1.0, 1.0], [x, y]),
-                        0.0
-                    ),
-                    MOI.LessThan{Float64}(3.0)
-                )
-            end
-        end
-        MOI.set(model, Xpress.CallbackFunction(), callback_function)
-        MOI.optimize!(model)
-        @test MOI.get(model, MOI.VariablePrimal(), x) == 1
-        @test MOI.get(model, MOI.VariablePrimal(), y) == 2
-        @test length(cb_calls) > 0
-    end
-    @testset "UserCut" begin
-        model, x, item_weights = callback_knapsack_model()
-        user_cut_submitted = false
-        cb_calls = Int32[]
-        MOI.set(model, Xpress.CallbackFunction(), (cb_data) -> begin
-            push!(cb_calls)
-            if Xpress.getintattrib(cb_data.model, Xpress.Lib.XPRS_CALLBACKCOUNT_OPTNODE) > 1
-                return
-            end
-            Xpress.get_cb_solution(model, cb_data.model)
-            terms = MOI.ScalarAffineTerm{Float64}[]
-            accumulated = 0.0
-            for (i, xi) in enumerate(x)
-                if MOI.get(model, MOI.CallbackVariablePrimal(cb_data), xi) > 0.0
-                    push!(terms, MOI.ScalarAffineTerm(1.0, xi))
-                    accumulated += item_weights[i]
-                end
-            end
-            if accumulated > 10.0
-                MOI.submit(
-                    model,
-                    MOI.UserCut(cb_data),
-                    MOI.ScalarAffineFunction{Float64}(terms, 0.0),
-                    MOI.LessThan{Float64}(length(terms) - 1)
-                )
-                user_cut_submitted = true
-            end
-            return
-        end)
-        MOI.optimize!(model)
-        @test user_cut_submitted
-    end
-    @testset "HeuristicSolution" begin
-        model, x, item_weights = callback_knapsack_model()
-        callback_called = false
-        cb_calls = Int32[]
-        MOI.set(model, Xpress.CallbackFunction(), (cb_data) -> begin
-            if Xpress.getintattrib(cb_data.model, Xpress.Lib.XPRS_CALLBACKCOUNT_OPTNODE) > 1
-                return
-            end
-            Xpress.get_cb_solution(model, cb_data.model)
-            x_vals = MOI.get.(model, MOI.CallbackVariablePrimal(cb_data), x)
-            @test MOI.submit(
-                model,
-                MOI.HeuristicSolution(cb_data),
-                x,
-                floor.(x_vals)
-            ) == MOI.HEURISTIC_SOLUTION_UNKNOWN
-            callback_called = true
-            return
-        end)
-        MOI.optimize!(model)
-        @test callback_called
-    end
-end
+# NOTE: 'Xpress.CallbackFunction' doesn't exist anymore
```

Comment:
We removed and dont have a replacement?
we should have a replacement

Reply:
We have a few replacements, actually (`OptNodeCallback`, `PreIntSolCallback`, `MessageCallback`)